model/itemTab.cc

```diff
@@ -148,10 +148,30 @@ namespace minsky
     auto minD=numeric_limits<float>::max();
     for (auto& i: itemCoords)
       {
-        float d=sqr((i.second).first+offsx-x)+sqr((i.second).second+offsy-y);
+        float xx=(i.second).first+offsx, yy=(i.second).second+offsy;  
+        float d=sqr(xx-x)+sqr(yy-y);
         float z=i.first->zoomFactor();
         float w=0.5*i.first->iWidth()*z,h=0.5*i.first->iHeight()*z;
-        if (d<minD && fabs((i.second).first+offsx-x)<w && fabs((i.second).second+offsy-y)<h)
+        // improve grabbing of Godley tables on the tab.
+        if (auto g=dynamic_pointer_cast<GodleyIcon>(i.first))
+          {
+            GodleyTableWindow godley(g);
```

Comment:
This seems a little heavyweight for itemAt, which is potentially called many times as a mouse is moved over the canvas.

I would have thought constructing the GodleyTableWindows and leaving them extent for the lifetime of the tab would be the way to go - that way they only need to be rendered when the tab is rendered, not every time the mouse moves.